packages/medusa/src/services/discount.ts

```diff
@@ -759,6 +773,35 @@ class DiscountService extends TransactionBaseService {
     return !!limit && count >= limit
   }
 
+  async hasCustomerReachedLimit(
+    discount: Discount,
+    customer: Customer
+  ): Promise<boolean> {
+    // If customer is empty on the cart, we will allow it
+    if (!customer) {
```

Comment:
check `if (!customer || !!limit) {` here to early return when the discount is not limited

Reply:
@DidierGuyon, we can add a check here for the limit and save a DB query.